test/e2e/generate/utils.go

```diff
@@ -0,0 +1,122 @@
+package generate
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"sigs.k8s.io/yaml"
+	"time"
+)
+
+type E2EClient struct {
+	Client dynamic.Interface
+}
+
+func NewE2EClient() (*E2EClient, error) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		kubeconfig = os.Getenv("HOME") + "/.kube/config"
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	e2eClient := new(E2EClient)
+	dClient, err := dynamic.NewForConfig(config)
+	e2eClient.Client = dClient
+	return e2eClient, err
+}
+
+// GetGVR :- gets GroupVersionResource for dynamic client
+func GetGVR(group, version, resource string) schema.GroupVersionResource {
+	return schema.GroupVersionResource{Group: group, Version: version, Resource: resource}
+}
+
+// CleanClusterPolicies ;- Deletes all the cluster policies
+func (e2e *E2EClient) CleanClusterPolicies(gvr schema.GroupVersionResource, namespace string) error {
```

Comment:
This `CleanClusterPolicies` function looks confused to me - the goal is the clean up cluster policies, while the implementation gets the namesapced resources and deletes them?

Reply:
The gvr that is passed to CleanClusterPolicies is kyverno gvr. i.e:- **clPolGVR = GetGVR("kyverno.io", "v1", "clusterpolicies")**.
And the namespace is Blank (""). So it will list and deletes ClusterPolicies.